### PR TITLE
Refs 768 : get list music by genre

### PR DESCRIPTION
**Pull request by yennguyen196**
https://edu-redmine.framgia.vn/issues/768

### CODE REVIEW

File: app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(mContext).inflate(R.layout.items_genre, null, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        final Genre genre = genres.get(position);
+        holder.bindView(genre, mListener);
+
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java]: remove bl

File: app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return genres.size();
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java]: check null

File: app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        private void bindView(final Genre genre, final OnGenreClickListener listener) {
+            mTextViewGenre.setText(genre.getName());
+            Glide.with(mImageViewGenre)
+                    .load(genre.getImageResource())
+                    .apply(new RequestOptions().override(100, 100)
+                            .placeholder(R.drawable.ic_launcher_background)
+                            .error(R.drawable.ic_launcher_background))
+                    .into(mImageViewGenre);
+            mImageViewGenre.setOnClickListener(new View.OnClickListener() {
```
Comment (tuannguyen38) [inline app/src/main/java/com/framgia/yen/mymusic/screen/main/home/GenreAdapter.java]: implement cho đẹp code

File: app/src/main/res/layout/fragment_genre_detail.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recycler_genre_detail"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+</android.support.constraint.ConstraintLayout>
```
Comment (tuannguyen38) [inline app/src/main/res/layout/fragment_genre_detail.xml]: enter last line

File: app/src/main/res/menu/menu_item_track.xml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item
+        android:id="@+id/action_download"
+        android:title="@string/string_download"/>
+    <item
+        android:id="@+id/action_add_to_favorite"
+        android:title="@string/string_add_to_favorite"/>
+</menu>
```
Comment (tuannguyen38) [inline app/src/main/res/menu/menu_item_track.xml]: enter last line

**Comment by DoanVanToan** [comment]
merged

**Pull request merged**